lib/Collection.js

```diff
@@ -120,4 +120,32 @@ module.exports = class Collection {
       {equals, has, capability, invocationSigner});
     return results;
   }
+
+  async countDocuments({id, type, equals, has}) {
+    if(!(id || type || equals || has)) {
+      throw new TypeError('"id", "type", "equals", or "has" must be given.');
+    }
+    if(!equals) {
+      equals = [];
+    } else if(!Array.isArray(equals)) {
+      equals = [equals];
+    } else {
+      equals = equals.slice();
+    }
+    if(id) {
+      equals.push({'content.id': id});
+    }
+    if(type) {
+      if(Array.isArray(type)) {
+        const query = type.map(type => ({'content.type': type}));
+        equals.push(...query);
+      } else {
+        equals.push({'content.type': type});
+      }
+    }
```

Comment:
Since this is the same code as in `find`, could you add an internal private function (name it something like `_prepareQuery` or anything better) at the bottom such that can be called in both places?